xptifw/basic_test/cl_processor.hpp

```diff
@@ -0,0 +1,620 @@
+//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//
+#pragma once
+#include "xpti_trace_framework.hpp"
+
+#include <chrono>
+#include <iomanip>
+#include <iostream>
+#include <map>
+#include <random>
+#include <set>
+#include <sstream>
+#include <stdint.h>
+#include <string>
+#include <tuple>
+#include <unordered_map>
+#include <vector>
+
+namespace test {
+namespace utils {
+enum class OptionType { Boolean, Integer, Float, String, Range };
+
+// We are using C++ 11, hence we cannot use
+// std::variant or std::any
+typedef std::map<int, long double> table_row_t;
+typedef std::map<int, table_row_t> table_t;
+typedef std::vector<std::string> titles_t;
+
+class ScopedTimer {
+public:
+  typedef std::chrono::time_point<std::chrono::high_resolution_clock>
+      time_unit_t;
+  ScopedTimer(uint64_t &ns, double &ratio, size_t count = 1)
+      : m_duration{ns}, m_average{ratio}, m_instances{count} {
+    m_before = std::chrono::high_resolution_clock::now();
+  }
+
+  ~ScopedTimer() {
+    m_after = std::chrono::high_resolution_clock::now();
+    auto duration = std::chrono::duration_cast<std::chrono::nanoseconds>(
+        m_after - m_before);
+    m_duration = duration.count();
+    m_average = (double)m_duration / m_instances;
+  }
+
+private:
+  uint64_t &m_duration;
```

Comment:
@tovinkere please, choose one style and stick to it.  Here you use snake_case, and a few lines later you use CamelCase.

```suggestion
  uint64_t &MDuration;
```

Reply:
@alexbatashev  The code went through a conversion from ISO style to the LLVM style and this is one of those missed ones. Will take care of it.

Reply:
This file has been addressed to use the prescribed convention